condor_history_to_prometheus.py

```diff
@@ -175,14 +174,19 @@ def read_from_schedd(schedd_ad, history=False, constraint='true', projection=[],
     prometheus_client.start_http_server(options.port)
 
     if options.collectors:
-        last_job = defaultdict(dict)
-        
-        for collector in args:
-            schedd_ads = locate_schedds(collector)
-            for s in schedd_ads:
-                last_job[s.get('Name')] = {'ClusterId': None, 'EnteredCurrentStatus': None}
+        last_job = last_jobs_dict(args)
+
+        if last_job is None:
+            logging.error(f'No schedds found')
+            exit()
 
         while True:
+            start = datetime.now()
             for collector in args:
                 query_collector(collector, metrics, last_job)
-            time.sleep(options.interval)
\ No newline at end of file
+
+            delta = datetime.now() - start
```

Comment:
If all you're interested in is a delta in seconds, `time.time()` is cheaper to run.